awx/main/models/oauth.py

```diff
@@ -146,5 +141,6 @@ def validate_external_users(self):
                 ).format(external_account))
 
     def save(self, *args, **kwargs):
-        self.validate_external_users()
```

Comment:
Only prevent OAuth2 tokens for external accounts *at creation time*.  Changing this means people can:

1.  `settings.ALLOW_OAUTH2_FOR_EXTERNAL_USERS = True`
2.  Create an OAuth2 token for a certain LDAP-backed user.
3.  `settings.ALLOW_OAUTH2_FOR_EXTERNAL_USERS = False`
4.  LDAP users can no longer make new OAuth2 tokens
5.  The token from step 2 will continue to function (until it's revoked).

Reply:
This lines up with how we explain its use in the docs and honestly makes tokens more useful for auditing in larger orgs.